pkg/sass-parser/lib/src/expression/number.test.ts

```diff
@@ -0,0 +1,210 @@
+      create: () => NumberExpression
+    ): void {
+      describe(description, () => {
+        beforeEach(() => void (node = create()));
+
+        it('has sassType number', () => expect(node.sassType).toBe('number'));
+
+        it('is a number', () => expect(node.value).toBe(123));
+
+        it('has a unit', () => expect(node.unit).toBe('px'));
+      });
+    }
+
+    describeNode('parsed', () => utils.parseExpression('123px'));
+
+    describeNode(
+      'constructed manually',
+      () =>
+        new NumberExpression({
+          value: 123,
+          unit: 'px',
+        })
+    );
+
+    describeNode('constructed from ExpressionProps', () =>
+      utils.fromExpressionProps({
+        value: 123,
+        unit: 'px',
+      })
+    );
+  });
+
+  describe('floating-point number', () => {
+    describe('unitless', () => {
+      beforeEach(() => void (node = utils.parseExpression('3.14')));
+
+      it('value', () => expect(node.value).toBe(3.14));
+
+      it('unit', () => expect(node.unit).toBeNull());
+    });
+
+    describe('with a unit', () => {
+      beforeEach(() => void (node = utils.parseExpression('1.618px')));
+
+      it('value', () => expect(node.value).toBe(1.618));
+
+      it('unit', () => expect(node.unit).toBe('px'));
+    });
+  });
+
+  describe('assigned new', () => {
+    beforeEach(() => void (node = utils.parseExpression('123')));
+
+    it('value', () => {
+      node.value = 456;
+      expect(node.value).toBe(456);
+    });
+
+    it('unit', () => {
+      node.unit = 'px';
+      expect(node.unit).toBe('px');
+    });
+  });
+
+  describe('stringifies', () => {
+    it('unitless', () =>
+      expect(utils.parseExpression('123').toString()).toBe('123'));
+
+    it('with a unit', () =>
+      expect(utils.parseExpression('123px').toString()).toBe('123px'));
+
+    it('floating-point number', () =>
+      expect(utils.parseExpression('3.14').toString()).toBe('3.14'));
+
+    describe('raws', () => {
+      it('with the same raw value as the expression', () =>
+        expect(
+          new NumberExpression({
+            value: 123,
+            raws: {number: {raw: 'hello', value: 123}},
+          }).toString()
+        ).toBe('hello'));
+
+      it('with a different raw value than the expression', () =>
+        expect(
+          new NumberExpression({
+            value: 123,
+            raws: {number: {raw: 'hello', value: 234}},
+          }).toString()
+        ).toBe('123'));
+
+      it('with a the same value as the expression but undefined format', () =>
+        expect(
+          new NumberExpression({
+            value: 123,
+            raws: {number: {raw: undefined, value: 123}},
+          }).toString()
+        ).toBe('123'));
+
+      it('with a the same value as the expression but falsy format', () =>
+        expect(
+          new NumberExpression({
+            value: 123,
+            raws: {number: {raw: '', value: 123}},
+          }).toString()
+        ).toBe(''));
```

Comment:
I don't know that these are really worth testing. The former should be forbidden by TypeScript and there's not a strong reason to think the latter would be an issue.